### PR TITLE
Digitigrade clothing fixes

### LINKED ISSUE #71221: Digitigrade variation of cargo tech shorts is not functional
## Reproduction:

image

image

### PR DESCRIPTION
**Pull request by lizardqueenlexi**
## About The Pull Request

Resolves #71221 and #71624.

Digitigrade uniform handling was broken for the single remaining uniform with a digitigrade variant, the cargo tech jumpsuit. Specifically, female characters would incorrectly display non-digitigrade shorts with digilegs, and vice versa. This PR rearranges some bodytype handling code to restore the correct behavior (it appears to have been broken by monkey clothing, of all things).

image

## Why It's Good For The Game

It looks bad when clothing does not line up with your character's sprite.
## Changelog
:cl:
fix: Made digitigrade clothing render correctly on female characters.
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/human/human_update_icons.dm [hunk 1]
```diff
@@ -105,11 +105,10 @@ There are several things that need to be remembered:
...
 			//BEGIN SPECIES HANDLING
-			if((dna?.species.bodytype & BODYTYPE_DIGITIGRADE) && (uniform.supports_variations_flags & CLOTHING_DIGITIGRADE_VARIATION))
-				icon_file = DIGITIGRADE_UNIFORM_FILE
 			if((dna?.species.bodytype & BODYTYPE_MONKEY) && (uniform.supports_variations_flags & CLOTHING_MONKEY_VARIATION))
 				icon_file = MONKEY_UNIFORM_FILE
-
+			else if((dna?.species.bodytype & BODYTYPE_DIGITIGRADE) && (uniform.supports_variations_flags & CLOTHING_DIGITIGRADE_VARIATION))
+				icon_file = DIGITIGRADE_UNIFORM_FILE
 			//Female sprites have lower priority than digitigrade sprites
 			else if(dna.species.sexes && (dna.species.bodytype & BODYTYPE_HUMANOID) && physique == FEMALE && !(uniform.female_sprite_flags & NO_FEMALE_UNIFORM)) //Agggggggghhhhh
```
Comment (Kapu1178) [inline code/modules/mob/living/carbon/human/human_update_icons.dm:113]: 🤔 two `else if`s isnt unreachable code?

also lol on my past comment being incorrect.

**Review by Kapu1178 — APPROVED** [review]

File: code/modules/mob/living/carbon/human/human_update_icons.dm (hunk 1 above)
Comment (lizardqueenlexi) [reply code/modules/mob/living/carbon/human/human_update_icons.dm:113]: Ah - that makes sense! In hindsight I'm not sure why I moved the digitigrade check to after the monkey one (all three are mutually exclusive anyway?) but the ordering works fine as long as the female check comes last.

Comment (Kapu1178) [reply code/modules/mob/living/carbon/human/human_update_icons.dm:113]: yes, they are mutually exclusive.

**Comment by RaveRadbury** [comment]
Not sure what you mean by underwear not rendering. I've got a digi leg character with underwear that shows up.

**Comment by lizardqueenlexi** [comment]
> Not sure what you mean by underwear not rendering. I've got a digi leg character with underwear that shows up.

This was poor phrasing on my part in the PR description - what I meant was that underwear sprites are shaped like normal legs and should _not_ appear if you have digitigrade ones.

**Comment by RaveRadbury** [comment]
> > Not sure what you mean by underwear not rendering. I've got a digi leg character with underwear that shows up.
> 
> This was poor phrasing on my part in the PR description - what I meant was that underwear sprites are shaped like normal legs and should _not_ appear if you have digitigrade ones.

Some underwear is not cut for the legs and stops before the leg type would be an issue?

**Comment by thgvr** [comment]
yeah uh, i'm not sure why you would disable underwear, you should not do that because digitigrade underwear sprites will never be acceped anywhere and i dont think people want ethereals v2 with no underwear allowed

**Comment by lizardqueenlexi** [comment]
> yeah uh, i'm not sure why you would disable underwear, you should not do that because digitigrade underwear sprites will never be acceped in tgstation and I don't think people want ethereals v2 with no underwear allowed. Most sprites fit it well enough anyways so why would it matter? really hope this doesnt go through with that lol

image

Boxers and long johns are the reason I disabled it, to be honest - I could also swear underwear _used_ to be disabled with digitigrade legs? It's not particularly important to this PR so I could easily remove that part. A way to disable _specific_ underwear types with the digitigrade legs would be ideal, but also probably not the easiest thing to implement.

**Comment by RaveRadbury** [comment]
> > yeah uh, i'm not sure why you would disable underwear, you should not do that because digitigrade underwear sprites will never be acceped in tgstation and I don't think people want ethereals v2 with no underwear allowed. Most sprites fit it well enough anyways so why would it matter? really hope this doesnt go through with that lol
> 
> image
> 
> Boxers and long johns are the reason I disabled it, to be honest - I could also swear underwear _used_ to be disabled with digitigrade legs? It's not particularly important to this PR so I could easily remove that part. A way to disable _specific_ underwear types with the digitigrade legs would be ideal, but also probably not the easiest thing to implement.

Seems like a big loss to throw out all the underwear that works for a small handful of ones that don't. Please reconsider this.

**Comment by thgvr** [comment]
You should just let it look like that, I would prefer not forcing every lizard to be naked without underwear because 2-3 of the sprites don't fit, and digitigrade options will never be accepted here.

**Comment by lizardqueenlexi** [comment]
Hm, yeah, I'm convinced enough regarding underwear. I didn't really like either solution, but in the absence of a much harder solution I think the option that leaves _most_ things looking alright is probably better. Thanks for the input!

**Pull request merged**